### DISCUSSION TITLE
Reticulum time intent

### DISCUSSION THREAD

**Discussion by faragher** [markqvist/Reticulum#280] [Help & Questions] [answered]
Looking at the readLoop function specifically, and I assume elsewhere, Reticulum uses time.time() to generate seconds from the epoch (more specifically it gets an integer with milliseconds) and while Python typically returns UTC, there's nothing directly specified in Python or even POSIX that specifies that.

However, I'm using TimeDateOffset in .net which allows an explicit call to get UTC. Is this the intent of the software or is it meant to be a local timezone? I'm going to assume it's UTC and continue working, but I want to be certain before we get some absurd edge case because I misunderstood the intent of the code.

**Comment by r8io** [comment]
Interestingly enough the Python docs talk about MOST systems: "On Windows and MOST Unix systems, the epoch is January 1, 1970, 00:00:00 (UTC) and leap seconds are not counted towards the time in seconds since the epoch"
https://docs.python.org/3.9/library/time.html#time.time

On my MacBook and the Debian VM running on my MacBook I get the same epoch time as from https://www.unixtimestamp.com, which is epoch time (UTC). So I would assume that my systems fall into the MOST category and return epoch time (UTC).

```
$ python
>>> import time
>>> print(time.time())
1682109041.468439
```

**Comment by ghost** [comment, marked as answer]
It's [Unix time](https://en.wikipedia.org/wiki/Unix_time), which is timezone-independent. It is defined as seconds since January 1, 1970 specifically in UTC. This means Unix time is the same for everyone in all timezones; convert to/from local time if needed.

**Reply by markqvist** [reply]
This is the answer :)

**Reply by faragher** [reply]
Perfect. Will explicitly target UTC, then.
